types/consensus_data.go

```diff
@@ -265,41 +265,79 @@ type AggregatorCommitteeConsensusData struct {
-	SyncCommitteeContributions []altair.SyncCommitteeContribution `ssz-max:"256"`
+	Contributors []AssignedAggregator `ssz-max:"2048"` // 512 * 4
+	// SyncCommitteeContributions is a list of contributions, one for each subcommittee
+	SyncCommitteeContributions []altair.SyncCommitteeContribution `ssz-max:"4"`
 }
 
 // Validate ensures the consensus data is internally consistent
 func (a *AggregatorCommitteeConsensusData) Validate() error {
-	// Aggregators validation
-	if len(a.Aggregators) != len(a.Attestations) {
-		return NewError(AggCommAggAttCntMismatchErrorCode, "aggregators and attestations count mismatch")
-	}
-	if len(a.Aggregators) != len(a.AggregatorsCommitteeIndexes) {
-		return NewError(AggCommAggCommIdxCntMismatchErrorCode, "aggregators and committee indexes count mismatch")
+
+	// Ensure at least one validator
+	if len(a.Aggregators) == 0 && len(a.Contributors) == 0 {
+		return NewError(AggCommConsensusDataNoValidatorErrorCode, "no validators assigned to aggregator committee or sync committee")
 	}
 
-	// Sync committee validation
-	if len(a.Contributors) != len(a.SyncCommitteeContributions) {
-		return NewError(AggCommContributorsContributionsCntMismatchErrorCode, "contributors and contributions count mismatch")
+	// Aggregators validation
+
+	// Ensure there is exactly one aggregated attestation per committee index
+	if len(a.AggregatorsCommitteeIndexes) != len(a.AggregatedAttestations) {
+		return NewError(AggCommAggCommIdxCntMismatchErrorCode, "committee indexes and attestations count mismatch")
 	}
 
-	// Optional: ensure all contributions reference only subnets listed in SyncCommitteeSubnets
-	validSubnets := make(map[uint64]struct{}, len(a.SyncCommitteeSubnets))
-	for _, s := range a.SyncCommitteeSubnets {
-		validSubnets[s] = struct{}{}
+	// Validate equal set (AggregatorsCommitteeIndexes vs. Aggregators.CommitteeIndex)
+	allowedAggCommittees := make(map[uint64]struct{}, len(a.AggregatorsCommitteeIndexes))
+	for _, idx := range a.AggregatorsCommitteeIndexes {
+		// Duplicates are not allowed
+		if _, dup := allowedAggCommittees[idx]; dup {
+			return NewError(AggCommDuplicatedCommIdxErrorCode, "duplicate index in AggregatorsCommitteeIndexes")
+		}
+		allowedAggCommittees[idx] = struct{}{}
 	}
+	usedAggCommittees := make(map[uint64]struct{}, len(a.AggregatorsCommitteeIndexes))
+	for _, agg := range a.Aggregators {
+		// Check it exists in allowed
+		if _, ok := allowedAggCommittees[agg.CommitteeIndex]; !ok {
+			return NewError(AggCommCommIdxMismatchErrorCode, "aggregator committee index not listed in AggregatorsCommitteeIndexes")
+		}
+		// Mark as used
+		usedAggCommittees[agg.CommitteeIndex] = struct{}{}
+	}
+	// Ensure no committee index was left unused (no more than necessary)
+	if len(usedAggCommittees) != len(allowedAggCommittees) {
+		return NewError(AggCommUnusedCommIdxErrorCode, "leftover aggregator committee index not usedAggCommittees by any aggregator")
+	}
```

Comment:
are there tests for the new errors?
will help me understand how they are all triggered